src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4035,7 +4041,10 @@ char* os::Linux::reserve_memory_special_huge_tlbfs_mixed(size_t bytes,
                                                          size_t alignment,
                                                          char* req_addr,
                                                          bool exec) {
-  size_t large_page_size = os::large_page_size();
+  // Select large_page_size from _page_sizes
+  // that is smaller than size_t bytes
+  size_t large_page_size = os::page_size_for_region_aligned(bytes, 1);
```

Comment:
This one also needs to use `os::page_size_for_region_unaligned(...)` since we know we have a size that needs both small and large pages.

Reply:
+1

Reply:
My comment here is a bit outdated. Now this function is used for both the "mixed" and the "only" case. As I see it we have to possibilities here. Either we stall this change a bit longer and let me finish the `ReservedSpace` work I've been doing lately. I have a patch coming out tonight or tomorrow that will add a page size member to `ReservedSpace`. I could then make a change to the `reserve_memory_special` API to pass a page size along with the other parameters and then this change will only be about enabling multiple page sizes. Everything else should just work. Alternatively we have to come up with some helper that decide the page size given the current alignment and size.